lib/src/metta/runner/stdlib/random.rs

```diff
@@ -0,0 +1,141 @@
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for RandomIntOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("random-int expects two arguments: number (start) and number (end)");
+        let start: i64 = args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?.into();
+        let end: i64 = args.get(1).and_then(Number::from_atom).ok_or_else(arg_error)?.into();
+        let range = start..end;
+        if range.is_empty() {
+            return Err(ExecError::from("Range is empty"));
+        }
+        let mut rng = rand::thread_rng();
+        Ok(vec![Atom::gnd(Number::Integer(rng.gen_range(range)))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct RandomFloatOp {}
+
+grounded_op!(RandomFloatOp, "random-float");
+
+impl Grounded for RandomFloatOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for RandomFloatOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("random-float expects two arguments: number (start) and number (end)");
+        let start: f64 = args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?.into();
+        let end: f64 = args.get(1).and_then(Number::from_atom).ok_or_else(arg_error)?.into();
+        let range = start..end;
+        if range.is_empty() {
+            return Err(ExecError::from("Range is empty"));
+        }
+        let mut rng = rand::thread_rng();
+        Ok(vec![Atom::gnd(Number::Float(rng.gen_range(range)))])
+    }
+}
+
+// NOTE: flip is absent in Python intentionally for conversion testing
+#[derive(Clone, PartialEq, Debug)]
+pub struct FlipOp{}
+
+impl Display for FlipOp {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "flip")
+    }
+}
+
+impl Grounded for FlipOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_BOOL])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for FlipOp {
+    fn execute(&self, _args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        Ok(vec![Atom::gnd(Bool(rand::random()))])
+    }
+}
+
+pub fn register_common_tokens(tref: &mut Tokenizer) {
+    let random_int_op = Atom::gnd(RandomIntOp{});
+    tref.register_token(regex(r"random-int"), move |_| { random_int_op.clone() });
+    let random_float_op = Atom::gnd(RandomFloatOp{});
+    tref.register_token(regex(r"random-float"), move |_| { random_float_op.clone() });
+}
+
+pub fn register_rust_stdlib_tokens(tref: &mut Tokenizer) {
+    let flip_op = Atom::gnd(FlipOp{});
+    tref.register_token(regex(r"flip"), move |_| { flip_op.clone() });
```

Comment:
Could be moved into `register_common_tokens` and `register_rust_stdlib_tokens` could be removed.